.github/workflows/deploy-infra.yaml

```diff
@@ -476,31 +470,36 @@ jobs:
         with:
           cmd: yq '.[env(ENVIRONMENT_NAME)].selfSignedCertificate' 'infra/config/github-environments-config.yaml'
         env:
-          ENVIRONMENT_NAME: ${{ matrix.environment-name }}
+          ENVIRONMENT_NAME: ${{ matrix.environment.name }}
 
       - id: fill-account-specific-metadata
         name: Fill in account specific metadata in ConfigMap
         run: |-
-          export DATADOG_API_KEY="${{ secrets.DATADOG_API_KEY }}"
-          export DATADOG_APP_KEY="${{ secrets.DATADOG_APP_KEY }}"
-
           yq -i '
-            .myAccount.domainName = strenv(DOMAIN_NAME) |
-            .myAccount.region = strenv(AWS_REGION) |
-            .myAccount.accountId = strenv(AWS_ACCOUNT_ID) |
-            .myAccount.publicFacing = strenv(PUBLIC_FACING) |
-            .myAccount.acmCertificate = strenv(ACM_CERTIFICATE_ARN) |
-            .myAccount.selfSignedCertificate = strenv(SELF_SIGNED_CERTIFICATE) |
-            .myAccount.datadogApiKey = strenv(DATADOG_API_KEY) |
-            .myAccount.datadogAppKey = strenv(DATADOG_APP_KEY)
-          ' infra/config/account-config.yaml
+              .myAccount.domainName = strenv(DOMAIN_NAME) |
+              .myAccount.region = strenv(AWS_REGION) |
+              .myAccount.accountId = strenv(AWS_ACCOUNT_ID) |
+              .myAccount.publicFacing = strenv(PUBLIC_FACING) |
+              .myAccount.acmCertificate = strenv(ACM_CERTIFICATE_ARN) |
+              .myAccount.selfSignedCertificate = strenv(SELF_SIGNED_CERTIFICATE)
+            ' infra/config/account-config.yaml
+
+          if [[ -n "${{ secrets.DATADOG_API_KEY }}" ]]
+          then
+            export DATADOG_API_KEY="${{ secrets.DATADOG_API_KEY }}"
+            export DATADOG_APP_KEY="${{ secrets.DATADOG_APP_KEY }}"
+            yq -i '
+              .myAccount.datadogAppKey = strenv(DATADOG_APP_KEY) |
+              .myAccount.datadogApiKey = strenv(DATADOG_API_KEY)
+            ' infra/config/account-config.yaml
+          fi
 
           cat infra/config/account-config.yaml
         env:
           AWS_REGION: ${{ secrets.AWS_REGION }}
           AWS_ACCOUNT_ID: ${{ steps.setup-aws.outputs.aws-account-id }}
           DOMAIN_NAME: ${{ steps.setup-domain.outputs.domain-name }}
-          ACM_CERTIFICATE_ARN: ${{ steps.setup-domain.outputs.acm-certificate-arn }}
+          ACM_CERTIFICATE_ARN: ${{ secrets.ACM_CERTIFICATE_ARN }}
```

Comment:
same as comment above:

setting this directly from the secret doesn't work for HMS staging as there are separate certificates for staging (`secrets.ACM_CERTIFICATE_ARN_STAGING`) and production (`secrets.ACM_CERTIFICATE_ARN`)